### PR TITLE
Add query for per-course-instance compute time

### PR DESCRIPTION
**Pull request by mwest1066**

### CODE REVIEW

**Review by nwalters512 — APPROVED** [review]
LGTM! Has this been tested with production data yet? I'm curious what kind of query execution durations to expect here.

File: admin_queries/course_instance_compute_time.sql [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        AND gj.date BETWEEN $start_date AND $end_date
+        AND gj.grading_method = 'External'
+    ORDER BY
+        ci.id, gj.id
+),
+total_grading_job_durations AS (
+    SELECT
+        course_instance_id,
+        COUNT(DISTINCT grading_job_id) AS grading_job_count,
+        extract(EPOCH FROM SUM(duration)) AS total_duration_hours
```
[the remark this answers is not in the document]
Comment (mwest1066) [reply admin_queries/course_instance_compute_time.sql:61]: Oh good catch! That was an untested last minute change, which are always the ones that introduce bugs!

**Comment by mwest1066** [comment]
> LGTM! Has this been tested with production data yet? I'm curious what kind of query execution durations to expect here.

"Yes" and "Long" 😭

On CA servers it takes several minutes (single digit minutes). I haven't dared to run it on US yet. I'd like to add duration tracking in the `workspaces` table so we don't need to reverse-engineer it from the logs.

**Comment by mwest1066** [comment]
I rewrote this PR to dynamically compute per-workspace usage and store it in the `workspaces` table. This simplifies (and speeds up) the admin query, as well as makes it easier to use this data in the future. The migrations now compute usage from the `workspace_logs` table and store it in `workspaces`.

File: admin_queries/course_instance_compute_time.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "description": "Course instances compute time within a given date range.",
+    "params": [
+        {
+            "name": "institution_short_name",
+            "description": "Restrict to a particular institution (blank means all institutions).",
+            "default": ""
+        },
+        {
+            "name": "start_date",
+            "description": "Start of the date range (e.g., first day of the academic yaer).",
```
Comment (nwalters512) [inline admin_queries/course_instance_compute_time.json:11]:
```suggestion
            "description": "Start of the date range (e.g., first day of the academic year).",
```

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**